src/Markdig/Helpers/CharHelper.cs

```diff
@@ -220,10 +220,16 @@ internal static bool IsSpaceOrPunctuation(this char c)
             }
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool IsNewLineLineFeed(this char c)
+        {
+            return c == '\n' || c == '\r';
+        }
+
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static bool IsNewLine(this char c)
```

Comment:
This is confusing to say that IsNewLine returns true for `\r`, so I would prefer that we don't change the semantic here. Use `IsNewLineOrLineFeed` in place where it needs to support `\r` and `\n`